pyop2/compilation.py

```diff
@@ -206,10 +220,34 @@ def get_so(self, src, extension):
                     # Atomically ensure soname exists
                     os.rename(tmpname, soname)
             # Wait for compilation to complete
+            newcomm.barrier()
             self.comm.barrier()
             # Load resulting library
             return ctypes.CDLL(soname)
 
+    @collective
+    def single_out_a_rank(self, comm):
+        rank = MPI.Comm.Get_rank(comm)
+        debug('Using file-based mechanism on %d', rank)
+        fname =  os.path.join("/tmp", "fb.%d" % rank)
```

Comment:
Previously, we used the configuration variable `"cache_dir"`, can you use that here?  Please also use `mkstemp` or whatever it's called to generate the prefix to the name (then broadcast it).